src/solver/simulation/sim_alloc_probleme_hebdo.cpp

```diff
@@ -336,7 +335,7 @@ void SIM_AllocationConstraints(PROBLEME_HEBDO& problem,
             break;
         }
         if (nbTimeSteps > 0)
-            problem.ResultatsContraintesCouplantes[k].variablesDuales.assign(nbTimeSteps, 0.);
+            problem.ResultatsContraintesCouplantes.emplace(bc, std::vector<double>(nbTimeSteps));
```

Comment:
`emplace` is variadic, you may give it the arguments you'd normally pass to the constructor
```cpp
template< class... Args >
std::pair<iterator, bool> emplace( Args&&... args );
```

It seems that the current code constructs a `std::vector`, then moves it. You can save the `std::move` step in this way.

```suggestion
            problem.ResultatsContraintesCouplantes.emplace(bc, nbTimeSteps, 0));
```

Also make sure to provide 0, otherwise the vector is unitialized.